include/hamlib/rig.h

```diff
@@ -2464,6 +2464,21 @@ typedef hamlib_port_t_deprecated port_t_deprecated;
 typedef hamlib_port_t port_t;
 #endif
 
+/* Macros to access port structures/pointers
+ * Make it easier to change location in preparation
+ *   for moving them out of rig->state.
+ * See https://github.com/Hamlib/Hamlib/issues/1445
+ */
+// Note: Experimental, and subject to change!!
+#define RIGPORT(r) (&r->state.rigport)
```

Comment:
This is a good start for fully transitioning to pointers in 5.0.

I'd try to prefix all definitions, both to scope them by device type and to be sure no name will easily conflict with any other code. Since these are specific rigs (radios), I'd just add a RIG_ prefix to all of them.

Have you checked rotator and amp structs BTW?